newsfeed-ionic/src/app/newsfeed/post/post-reaction/post-reaction.component.ts

```diff
@@ -1,38 +1,44 @@
 import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
-import { ReactionTypes } from 'src/app/interfaces/reactionTypes';
+import { ReactionType } from 'src/app/enums/reaction-type';
 import { NewsfeedDataService } from 'src/app/services/newsfeed-data.service';
 import { IComment } from 'src/app/interfaces/icomment';
+import { IPost } from 'src/app/interfaces/ipost';
+import { IReaction } from 'src/app/interfaces/ireaction';
 
 @Component({
   selector: 'app-post-reaction',
   templateUrl: './post-reaction.component.html',
   styleUrls: ['./post-reaction.component.scss'],
 })
 export class PostReactionComponent implements OnInit {
-  constructor(private newsfeedDataService: NewsfeedDataService) {}
+  @Input() post!: IPost;
+  @Input() numberOfLikes = 0;
+  @Input() numberOfHearts = 0;
+  @Input() numberOfSmiles = 0;
 
-  @Input() postId = 0;
-  comments: IComment[] = [];
-  numberOfLikes = 0;
-  numberOfHearts = 0;
-  numberOfSmiles = 0;
+  constructor(private newsfeedDataService: NewsfeedDataService) {}
 
   ngOnInit() {}
 
-  reactToPost(clickEvent: any, reactionType: number) {
-    switch (reactionType) {
-      case ReactionTypes.Like:
-        this.newsfeedDataService.addReaction(this.postId, this.newsfeedDataService.defaultUser.id, ReactionTypes.Like);
-        this.numberOfLikes = this.newsfeedDataService.getNumberOfLikes(this.postId);
-        break;
-      case ReactionTypes.Heart:
-        this.newsfeedDataService.addReaction(this.postId, this.newsfeedDataService.defaultUser.id, ReactionTypes.Heart);
-        this.numberOfHearts = this.newsfeedDataService.getNumberOfHearts(this.postId);
-        break;
-      case ReactionTypes.Smile:
-        this.newsfeedDataService.addReaction(this.postId, this.newsfeedDataService.defaultUser.id, ReactionTypes.Smile);
-        this.numberOfSmiles = this.newsfeedDataService.getNumberOfSmiles(this.postId);
-        break;
-    }
+  reactToPost(clickEvent: any, reactionType: ReactionType) {
+    this.newsfeedDataService.addReaction(this.post.id, reactionType);
+
+    const reactions: IReaction[] = this.post.reactions;
+
+    this.numberOfLikes = this.newsfeedDataService.getNumberOfLikes(reactions);
+    this.numberOfHearts = this.newsfeedDataService.getNumberOfHearts(reactions);
+    this.numberOfSmiles = this.newsfeedDataService.getNumberOfSmiles(reactions);
+  }
+
+  get LIKE(): ReactionType {
+    return ReactionType.LIKE;
+  }
+
+  get HEART(): ReactionType {
+    return ReactionType.HEART;
+  }
+
+  get SMILE(): ReactionType {
+    return ReactionType.SMILE;
```

Comment:
Or you could instead bind `ReactionType` directly to the component and use that on the template

```ts
class MyComponent {
  reactionType = ReactionType
}
```